lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,217 @@ def load_team_membership_csv(course, response):
+                    # if a team doesn't exists, the validation doesn't apply to it.
+                    if user.id in self.user_ids_by_teamset_index[i] and self.add_error_and_check_if_max_exceeded(
+                            'User ' + user.id.__str__() + ' is already on a team set.'):
+                        return False
+                    else:
+                        self.user_ids_by_teamset_index[i].add(user.id)
+                        continue
+                max_team_size = self.course_module.teams_configuration.default_max_team_size
+                if max_team_size is not None and team.users.count() >= max_team_size:
+                    if self.add_error_and_check_if_max_exceeded('Team ' + team.team_id + ' is already full.'):
+                        return False
+                if CourseTeamMembership.user_in_team_for_course(user, self.course.id, team.topic_id):
+                    if self.add_error_and_check_if_max_exceeded(
+                        'The user ' + user.username + ' is already a member of a team inside teamset '
+                        + team.topic_id + ' in this course.'
+                    ):
+                        return False
+
+    def add_error_and_check_if_max_exceeded(self, error_message):
+        """
+        Adds an error to the error collection.
+        :param error_message:
+        :return: True if maximum error threshold is exceeded and processing must stop
+                 False if maximum error threshold is NOT exceeded and processing can continue
+        """
+        self.validation_errors.append(error_message)
+        if len(self.validation_errors) >= self.max_erros:
+            self.max_erros_found = True
+            return True
+        else:
+            return False
+
+    def add_user_to_team(self, user_row):
+        """
+        Creates a CourseTeamMembership entry - i.e: a relationship between a user and a team.
+        user_row is the list representation of an input row. It will have the following formta:
+        use_id, enrollment_mode, <Team_Name_1>,...,<Team_Name_n>
+        Team_Name_x are optional and can be a sparse list i.e:
+        andrew,masters,team1,,team3
+        joe,masters,,team2,team3
+        """
+        user = user_row[0]
+        for i in range(2, len(user_row)):
+            team_name = user_row[i]
+            if team_name:
+                try:
+                    # checks for a team inside a specific team set. This way team names can be duplicated across
+                    # teamsets
+                    team = CourseTeam.objects.get(name=team_name, topic_id=self.teamset_name_by_index[i])
+                except CourseTeam.DoesNotExist:
+                    team = CourseTeam.create(name=team_name, course_id=self.course.id, description='Import from csv',
+                                             topic_id=self.teamset_name_by_index[i]
+                                             )
+                    team.save()
```

Comment:
If you do `CourseTeam.objects.create(...)` instead, that will save the object for you, without you having to explicitly call `team.save()`.